Userland/Libraries/LibWeb/WebGL/WebGLContextAttributes.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022, Luke Wilde <lukew@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/Completion.h>
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibWeb/WebGL/WebGLContextAttributes.h>
+
+namespace Web::WebGL {
+
+JS::ThrowCompletionOr<WebGLContextAttributes> convert_value_to_context_attributes_dictionary(JS::GlobalObject& global_object, JS::Value value)
+{
+    auto& vm = global_object.vm();
+
+    // NOTE: This code was generated by the IDL code generator and then cleaned up.
```

Comment:
Hmm, why can we not generate it at build time then? :sweat_smile:

Reply:
We can generate it at build time, it's that it's generated when something (e.g. an argument) has a dictionary type. Since getContext takes `any` as the options, we can't convert it to the dictionary until we know which context (and thus which dictionary it is) we are creating.

The solution to this is to generate conversion functions for the dictionaries that can be accessed globally. I'm not sure where to put this though. Maybe in the Wrapper.h of the interface the dictionary is declared next to?

Reply:
I see - happy to keep it for now and maybe we can replace it with build time generated code later. :+1: